### ISSUE TITLE
Using `fix` on symbol defined over option types in output causes strange error

### ISSUE THREAD

**Issue by miquelramirez** [MiniZinc/libminizinc#517]
Hello,

while discussing some modelling ideas with @pstuckey we came up with the `minizinc` model below. I am using version 2.5.5/273051464. 

Note that the variable `above` is defined using the `option type` feature:

```
% The blocks
enum BLOCKS;
BLOCKS = { A, B, C, D };
% The surfaces blocks rest on which are not blocks
enum SURFACES;
SURFACES = { table };
% Stages/happenings required to specify the action
enum STAGES;
STAGES = {s0, s1};

% State variables and their domains

% holding(b)
array[BLOCKS, STAGES] of var 0..1: holding;
% on(b,b') becomes above(b, b')
array[BLOCKS, STAGES] of var opt BLOCKS: above;
% we keep the on(b, s) for blocks on surfaces
array[BLOCKS, SURFACES, STAGES] of var 0..1: on;

% Action variable
var BLOCKS: target_block;
var SURFACES: target_surface;

% DOMAIN CONSTRAINTS - common to all instances and actions
% the clear axiom

%predicate clear(var BLOCKS: b, STAGES: i) =
%   absent(above[b,i]) /\ holding[b,i] = 0;
      
% the handempty axiom
predicate handempty(STAGES: i) =
  forall(b in BLOCKS)(holding[b, i]=0);
  
% Invariant MiniZinc/MiniZincIDE#1: only one block can be hold at a time  
predicate gripper_capacity(STAGES: i)=
  sum([holding[b, i] | b in BLOCKS]) <= 1;
  
% Invariant MiniZinc/MiniZincIDE#2: blocks need to be held or lying on some other object
predicate block_location(STAGES: i)=
  forall (b in BLOCKS) (holding[b,i] + sum(s in SURFACES)(on[b,s,i]) + sum(b2 in BLOCKS)(above[b2,i]=b)=1);
    
%% PICK UP ACTION CONSTRAINTS

%% CHANGE CONSTRAINT

constraint on[target_block, target_surface, s0] = 1 /\ on[target_block, target_surface, s1] = 0;
constraint holding[target_block, s0] = 0 /\ holding[target_block, s1] = 1;
constraint above[target_block, s0] = <> /\ above[target_block, s1] = <>;


%% INVARIANTS

%% DOMAIN
constraint gripper_capacity(s0) /\ gripper_capacity(s1);
constraint block_location(s0) /\ block_location(s1);
%% ACTION SPECIFIC
%% constraint clear(target_block, s0) /\ not clear(target_block, s1);
constraint handempty(s0) /\ not handempty(s1);

%% FRAME (these are DOMAIN constraints)
constraint forall(b in BLOCKS)(if target_block != b then holding[b, s0] = holding[b, s1] else true endif);

constraint forall(b in BLOCKS)(
              forall(s in SURFACES)(
                  if target_block != b then on[b, s, s0] = on[b, s, s1] else true endif
              )
           );
           
constraint forall(b in BLOCKS)(
                above[b, s0] = above[b, s1]
            );

solve satisfy;

output[
  "target_block=\(target_block), target_surface=\(target_surface)\n"
] 
++ [ "TNF changes\n" ]
++ [ "holding(\(b)): \(holding[b, s0]) -> \(holding[b, s1])\n" | b in BLOCKS]
++ [ "above(\(b)): \(above[b, s0]) -> \(above[b, s1])\n" | b in BLOCKS where fix(above[b, s0]) != fix(above[b, s1])]
++ [ "on(\(b),\(s)): \(on[b, s, s0]) -> \(on[b, s, s1])\n" | b in BLOCKS, s in SURFACES];
```

When running this model through the Minizinc IDE I am getting the following error:

```
Running pickup_v2.mzn
unknown file:0.0:
MiniZinc: evaluation error: undeclared identifier '_absent'
Process finished with non-zero exit code 1
Finished in 156msec
```

If the line 
```
++ [ "above(\(b)): \(above[b, s0]) -> \(above[b, s1])\n" | b in BLOCKS where fix(above[b, s0]) != fix(above[b, s1])]
```

is replaced with:

```
++ [ "above(\(b)): \(above[b, s0]) -> \(above[b, s1])\n" | b in BLOCKS]
```

the issue goes away.

Note that every occurrence of the `absent` function are commented out.

I hope this helps,

Miquel.

**Comment by guidotack** [comment]
This issue has already been fixed in b2a3d435a39067c1b35e877c030736ae36b0969f. The fix will be part of the next release.

**Comment by miquelramirez** [comment]
Thanks @guidotack !
